packages/babel-types/src/converters/valueToNode.js

```diff
@@ -1,6 +1,6 @@
 // @flow
 import isPlainObject from "lodash/isPlainObject";
-import isRegExp from "lodash/isRegExp";
+import isRegExp from "is-regexp";
```

Comment:
`_.isRegExp` will call optimized node.js builtin `util.isRegExp` before `toString` comparison. Given that most Babel users are using node.js, I would suggest in this case we create our own `isRegExp` helper.
```js
function isRegExp(value) {
  try {
    return require("util").isRegExp(value);
  } catch {
    return Object.prototype.toString.call(value) === '[object RegExp]';
  }
}

Reply:
Oh `util.isRegExp` is deprecated, maybe we should target this change to Babel 8 via https://nodejs.org/api/util.html#util_util_types_isregexp_value.

Reply:
Is it worth somehow pushing the `require` operation to a higher (module?) scope so that the `try...require` is only evaluated once?

Reply:
Ah, I missed your second comment before replying.  Yep, I'll retarget this against Babel 8 when applying a change to use a built-in NodeJS regexp test.

Reply:
> Is it worth somehow pushing the require operation to a higher (module?) scope so that the try...require is only evaluated once?

Good idea, we can even get rid of try catch here 
```
var nodeUtil = (function() {
  try {
    return require('util');
  } catch {}
}());

isRegExp = value =>
   nodeUtil ? nodeUtil.isRegExp(value) : Object.prototype.toString.call(value) === '[object RegExp]';
```

Reply:
Hrm.  After a little more time to pause I'm less certain what to do here.  The `process.binding` pattern doesn't appear elsewhere in the codebase, and although `nodejs` may be relatively well-adopted by `babel` users, I don't see too many references to platform-specific code (just [one](https://github.com/babel/babel/blob/e2b33c2cb3272cf46d4ca29097dc331d7570fca5/packages/babel-core/src/config/files/module-types.js#L60) or [two](https://github.com/babel/babel/blob/e2b33c2cb3272cf46d4ca29097dc331d7570fca5/packages/babel-register/src/cache.js#L45) mentions of nodejs-related workarounds).

I'm not averse to applying those changes if they're definitely the way to go, but I'd prefer to understand a bit more about the options and implications first.

Reply:
Note that `process.binding` is also deprecated (see [DEP0103](https://nodejs.org/dist/latest/docs/api/deprecations.html#deprecations_dep0103_process_binding_util_is_typechecks) and [DEP0111](https://nodejs.org/dist/latest/docs/api/deprecations.html#DEP0111)).  For using Node.js methods I would probably go for:
```js
const nodeUtilTypes = util.types || util;
```

This way you would use `util.types.isRegExp` in Node.js 10+, `util.isRegExp` in older versions.  Although `util.isRegExp` is deprecated in Node.js 4+ it's never going to be removed or produce a warning in Node.js < 10 so it seems like a reasonable fallback.

Reply:
As of `process.binding()`, I meant to avoid interactions with bundlers -- that `require("utils")` is intended to be failed on browsers. Since we have offered a fallback, we don't expect the bundlers do anything about `require("utils")` So users don't have to provide `node: { utils: empty }` to Webpack or use `rollup-plugin-node-polyfills` that may even try to polyfill `utils.isRegExp`.

As a note, `@babel/standalone` bundles `@babel/types` and it is run in browsers. Building `@babel/standalone` does not fail because we have `rollup-plugin-node-polyfills`, but I don't expect `@babel/types` _bundling_ users have to change their bundler config in a non-breaking releases.

Reply:
@JLHwung @coreyfarrell The code structure has been refactored a bunch since the previous comments in this discussion; in particular the top-level `util` module is always imported, and then a local `isRegExp` helper is assigned based on a series of preferentially-ordered fallbacks.  Hopefully that meets most of the optimization goals mentioned while also catering for older Node versions; glad for further feedback.